cli/admin/src/test/java/io/pravega/cli/admin/controller/AbstractControllerMetadataCommandsTest.java

```diff
@@ -343,6 +354,55 @@ public void testControllerMetadataViewReaderInfoCommandWithException() throws Ex
     }
 
+    private CreateReaderGroupEvent buildCreateRGEvent(String scope, String rgName, ReaderGroupConfig config,
+                                                      final long requestId, final long createTimestamp) {
+        Map<String, RGStreamCutRecord> startStreamCuts = config.getStartingStreamCuts().entrySet().stream()
+                                                               .collect(Collectors.toMap(e -> e.getKey().getScopedName(),
+                                                                       e -> new RGStreamCutRecord(ImmutableMap.copyOf(ModelHelper.getStreamCutMap(e.getValue())))));
+        Map<String, RGStreamCutRecord> endStreamCuts = config.getEndingStreamCuts().entrySet().stream()
+                                                             .collect(Collectors.toMap(e -> e.getKey().getScopedName(),
+                                                                     e -> new RGStreamCutRecord(ImmutableMap.copyOf(ModelHelper.getStreamCutMap(e.getValue())))));
+        return new CreateReaderGroupEvent(requestId, scope, rgName, config.getGroupRefreshTimeMillis(),
+                config.getAutomaticCheckpointIntervalMillis(), config.getMaxOutstandingCheckpointRequest(),
+                config.getRetentionType().ordinal(), config.getGeneration(), config.getReaderGroupId(),
+                startStreamCuts, endStreamCuts, createTimestamp);
+    }
+
+    @Test
+    public void testControllerMetadataViewPendingEventsCommand() throws Exception {
+        final String host = UUID.randomUUID().toString();
+        final String requestId = UUID.randomUUID().toString();
+
+        @Cleanup("shutdownNow")
+        ScheduledExecutorService executorService = ExecutorServiceHelpers.newScheduledThreadPool(1, "cli-test");
+        ZKHelper zkHelper = ZKHelper.create(SETUP_UTILS.getZkTestServer().getConnectString(), "pravega-cluster");
+        ZKHostIndex hostIndex = zkHelper.getZkHostIndex(executorService);
+        EventHelper eventHelper = new EventHelper(executorService, host, hostIndex);
+
+        String scopedStreamName = "scope/stream";
+        ReaderGroupConfig rgConf = ReaderGroupConfig.builder().disableAutomaticCheckpoints()
+                                                    .stream(scopedStreamName)
+                                                    .retentionType(ReaderGroupConfig.StreamDataRetention.NONE)
+                                                    .build();
+        CreateReaderGroupEvent createEventOne = buildCreateRGEvent("scope", "rg1", rgConf, 1L, System.currentTimeMillis());
+
+        eventHelper.addRequestToIndex(host, requestId, createEventOne);
+        String commandResult = TestUtils.executeCommand("controller-metadata request-detail " + host + " " + requestId, STATE.get());
+        Assert.assertTrue(commandResult.contains("rg1"));
+
+        //No metadata found scenario.
+        byte[] emptyByteData = new byte[0];
+        hostIndex.removeEntity(host, requestId, false);
```

Comment:
Same happens here with the next 2 calls.

Reply:
ZKHostIndexis not closable.